sdk/lib/rtl/rangelist.c

```diff
@@ -13,16 +13,478 @@
+
+    return STATUS_SUCCESS;
+}
+
+NTSTATUS
+NTAPI 
+RtlpAddIntersectingRanges(
+    _In_ PLIST_ENTRY ListHead,
+    _In_ PRTLP_RANGE_LIST_ENTRY RtlEntry,
+    _In_ PRTLP_RANGE_LIST_ENTRY AddRtlEntry,
+    _In_ ULONG Flags)
+{
+    PRTLP_RANGE_LIST_ENTRY CurrentRtlEntry;
+    PRTLP_RANGE_LIST_ENTRY NextRtlEntry;
+    PRTLP_RANGE_LIST_ENTRY MergedRtlEntry;
+    PRTLP_RANGE_LIST_ENTRY NextMergedRtlEntry;
+    NTSTATUS Status;
+    BOOLEAN AddRtlEntryIsShared;
+    BOOLEAN MergedRtlEntryIsShared;
+
+    PAGED_CODE_RTL();
+    DPRINT("RtlpAddIntersectingRanges: [%X], %p, %p [%I64X-%I64X], %p [%I64X-%I64X]\n", Flags, ListHead, RtlEntry, RtlEntry->Start, RtlEntry->End, AddRtlEntry, AddRtlEntry->Start, AddRtlEntry->End);
+
+    ASSERT(RtlEntry);
+    ASSERT(AddRtlEntry);
+
+    AddRtlEntryIsShared = (AddRtlEntry->PublicFlags & RTL_RANGE_SHARED);
+
+    if ((Flags & RTL_RANGE_LIST_ADD_IF_CONFLICT) == 0)
+    {
+        for(CurrentRtlEntry = RtlEntry;
+            &CurrentRtlEntry->ListEntry != ListHead;
+            CurrentRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->ListEntry.Flink))
+        {
+            if (AddRtlEntry->End < CurrentRtlEntry->Start) {
+                break;
+            }
+
+            if (CurrentRtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED)
+            {
+                for (MergedRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->Merged.ListHead.Flink);
+                     &MergedRtlEntry->ListEntry != &CurrentRtlEntry->Merged.ListHead;
+                     MergedRtlEntry = RtlpEntryFromLink(MergedRtlEntry->ListEntry.Flink))
+                {
+                    MergedRtlEntryIsShared = (MergedRtlEntry->PublicFlags & RTL_RANGE_SHARED);
+
+                    if (IsRangesIntersection(MergedRtlEntry, AddRtlEntry) &&
+                        !(AddRtlEntryIsShared && MergedRtlEntryIsShared))
+                    {
+                        DPRINT1("RtlpAddIntersectingRanges: STATUS_RANGE_LIST_CONFLICT\n");
+                        return STATUS_RANGE_LIST_CONFLICT;
+                    }
+                }
+            }
+            else if (!AddRtlEntryIsShared ||
+                     !(CurrentRtlEntry->PublicFlags & RTL_RANGE_SHARED))
+            {
+                DPRINT1("RtlpAddIntersectingRanges: STATUS_RANGE_LIST_CONFLICT\n");
+                return STATUS_RANGE_LIST_CONFLICT;
+            }
+        }
+    }
+
+    if (!(RtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED))
+    {
+        Status = RtlpConvertToMergedRange(RtlEntry);
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("RtlpAddIntersectingRanges: Status %X\n", Status);
+            return Status;
+        }
+    }
+
+    ASSERT((RtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED));
+
+    CurrentRtlEntry = RtlpEntryFromLink(RtlEntry->ListEntry.Flink);
+    NextRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->ListEntry.Flink);
+
+    while (&CurrentRtlEntry->ListEntry != ListHead)
```

Comment:
Here you use `while` loop for walking through the list, above you use `for` loop. Please, be consistent :)

Reply:
In fact, I am consistent. 
For simple cases I used "for" loops
For safe (CurrentEntry + NextEntry) cases - "while" loops.
There are code examples above.